### PR TITLE
Update pinejs and make use of named keys

### PR DESCRIPTION
**Pull request by Page-**

### CODE REVIEW

File: src/routes/devices.ts [hunk 1]
```diff
@@ -662,11 +662,13 @@ const upsertImageInstall = async (
 		}
 		await resinApi.patch({
 			resource: 'image_install',
+			id: {
+				installs__image: imageId,
+				device: deviceId,
```
Comment (thgreasi) [inline src/routes/devices.ts:667]:
```suggestion
				device: deviceId,
				installs__image: imageId,
```
Reordered to match the sbvr :)

File: src/routes/devices.ts [hunk 2]
```diff
@@ -705,16 +707,14 @@ const upsertGatewayDownload = async (
 	} else {
 		await resinApi.patch({
 			resource: 'gateway_download',
+			id: {
+				is_downloaded_by__device: deviceId,
+				image: imageId,
```
Comment (thgreasi) [inline src/routes/devices.ts:712]:
```suggestion
				image: imageId,
				is_downloaded_by__device: deviceId,
```
Reorder to match the sbvr

File: src/routes/registry.ts [hunk 3]
```diff
@@ -341,7 +343,7 @@ const $getSubject = memoize(
 							},
 						},
 					},
-				})) as AnyObject[];
+				})) as AnyObject;
 				if (device != null && device.id != null) {
```
Comment (thgreasi) [inline src/routes/registry.ts:347]: Aside: Seems we don't actually need to check the `device.id` and can drop it.

File: src/routes/registry.ts [hunk 4]
```diff
@@ -316,14 +316,16 @@ const $getSubject = memoize(
 		if (subject) {
 			try {
 				// Try to resolve as a device api key first, using the passed in subject
-				const [device] = (await api.resin.get({
+				const device = (await api.resin.get({
 					resource: 'device',
 					passthrough: { req: permissions.root },
+					id: {
+						// uuids are passed as `d_${uuid}`
+						uuid: subject.replace(/^d_/, ''),
```
Comment (thgreasi) [inline src/routes/registry.ts:324]: Not against it, but wanted to say that having the natural key in the `id` as well as a `$filter` a bit lower felt a bit disconnected. 
Maybe it's just the part of me which thinks/translates $filter as a WHERE...

File: src/routes/devices.ts [hunk 5]
```diff
@@ -626,16 +626,16 @@ const upsertImageInstall = async (
 	releaseId: number,
 	dlProg?: number,
 ): Promise<void> => {
-	const [imgInstall] = (await resinApi.get({
+	const imgInstall = (await resinApi.get({
 		resource: 'image_install',
+		id: {
+			installs__image: imageId,
```
Comment (thgreasi) [inline src/routes/devices.ts:632]: BTW, will pine error if we specify an incomplete natural key or props that non-unique props in the `id` field?
Eg: just the `id: {device}`, or `id: {device, created_at}`

File: src/routes/registry.ts (hunk 4 above)
Comment (Page-) [reply src/routes/registry.ts:324]: It's not in the $filter? The only places it should still be in the $filter if it's something like `uuid: { $in: ... }` or `uuid: { $startswith: '...' }`

File: src/routes/devices.ts (hunk 5 above)
Comment (Page-) [reply src/routes/devices.ts:632]: Yes

Comment (thgreasi) [reply src/routes/devices.ts:632]: Awesome?

**Pull request merged**